### DISCUSSION TITLE
CommonJS module

### DISCUSSION THREAD

**Discussion by torres-developer** [Myndex/apca-w3#8] [Q&A] [answered]
I want to create an [PostCSS](https://postcss.org/) plugin that it's like a polyfill for new CSS functions that will help choosing the right contrast color and things like that. The problem is that I think it's only possible to create a plugin using CommonJS modules. I'm not an expert in JavaScript so I don't know if there's a workaround to transform ES6 modules to CommonJS modules. I was thinking about just copying the [apca-w3.js file](https://github.com/Myndex/apca-w3/blob/master/src/apca-w3.js) and just turn it into a CommonJS module but I don't know if it is possible because of the licensing. What should I do?

It possible to create an npm package that comes with the same code but in CommonJS?

Thank you!

**Comment by Myndex** [comment]
Hi @torres-developer  thank you for writing.

We are working on adding build tools so various flavors will be generated. 

If your plug in commercial or open source?

**Reply by torres-developer** [reply, marked as answer]
Hello again!

I asked for help on a place for support/discussion on PostCSS and this is the conversation.

![gitter im_postcss_postcss](https://user-images.githubusercontent.com/110784622/185602375-46dd11dc-fcaa-4e3e-8193-963695535ddb.png)

Because he said he wouldn't recommend it I assumed I can do it anyways. I looked the source code for PostCSS and I figured out how I can create the plugin using ES6 modules.

[The plugin is open source](https://github.com/torres-developer/postcss-accessibility). It's still in development but I think it's working good for now. I'm trying to use this library right so I don't generate wrong results.

Looking forward to make the plugin easy to use so people can easily write CSS where you just give the background color and it automatically chooses a good contrast color for the text depending on the text size, weight and usage (fluent, sub-fluent, ...) that the designer wants the text to have.

Here is an example:

```css
p {
    background-color: hsl(192 62% 94%);
    color: a11y-txt(hsl(192 62% 94%), 1.75em, 400);
    font-size: 1.75em;
}
```

The a11y-txt function needs at least 1 parameter (the background-color value). Then you also can give a font-size (default 16px), a font-weight (default 400) and a usage (default bodyText). Probably needs a better name the function. After compiling the CSS above with the plugin, the CSS becomes:

```css
p {
    background-color: hsl(192 62% 94%);
    color: hsl(192.63157894736838 61.290322580645174% 17%);
    font-size: 1.75em;
}
```

Probably need too remove the decimal part of those numbers.

[myndex.com/SAPC with the colors above.](https://www.myndex.com/SAPC/?BG=113a46&TXT=e6f5f9)

The only problem is that isn't compatible for people that use CommonJS for their config but it's very easy to make it ES6.

So I don't think I need a CommonJS module of this repository anymore :P Sorry for bothering.

**Comment by caseyjhol** [comment]
@Myndex Any chance of the BuildTools branch ever getting published to NPM (even if as a beta or using the "next" tag)?

**Reply by Myndex** [reply]
it's a bit of a project, and I have not had time to get to it... but volunteers are welcome 😌

**Reply by caseyjhol** [reply]
Unfortunately, due to the somewhat confusing nature of the license, I had to switch to using the implementation in https://github.com/color-js/color.js

**Reply by Myndex** [reply]
This will be resolved sooner rather than later
